### PR TITLE
[env/ohbm] add category/subcategory filtering to PosterHall

### PR DESCRIPTION
**Pull request by margulies**
This PR contains filtering on the category and subcategory level. It is highly specific to env/ohbm, where a postervenue has multiple category each with subcategories organized hierarchically. The category and subcategory filters are integrated with the isLive and Bookmark filters as well.

### CODE REVIEW

**Comment by yarikoptic** [comment]
it would close https://github.com/sparkletown/sparkle/issues/1576 AFAIK! ;)  Well done!!!

**Comment by yarikoptic** [comment]
most likely would need "integration" with https://github.com/sparkletown/sparkle/pull/1557 by @goran-peoski-work . Might as well be worthwhile to get ready and prepare a merge if (hopefully) that PR gets merged soon ;)

File: src/settings.ts [hunk 1]
```diff
@@ -154,7 +154,7 @@ export const PLAYA_VENUE_STYLES: Record<string, CSSProperties> = {
 };
 
 export const ACCEPTED_IMAGE_TYPES =
-  "image/png,image/x-png,image/gif,image/jpg,image/jpeg,image/tiff,image/bmp,image/gif";
+  "image/png,image/x-png,image/gif,image/jpg,image/jpeg,image/tiff,image/bmp,image/gif,image/webp";
```
Comment (yarikoptic) [inline src/settings.ts:157]: May be with submitting this commit as a separate PR right away against staging?

Comment (margulies) [reply src/settings.ts:157]: Good call -- thanks! https://github.com/sparkletown/sparkle/pull/1601

**Comment by sparkleed** [comment]
This improvement to the UX has been requested by OHBM, on basis that it dramatically improves UX of finding posters (and makes up for the fact that search results on title key terms currently present in reverse-order, an issue for which there is no obvious current understanding or fix.

The code has been run by different people and different devices, and has been correspondingly QA-ed.

@margulies has said that the code, though carefully structured and limited to this context, may not be of truly Sparkle-generality and quality; I'd nonetheless like to see this merged as a first priority overnight - for among other reasons that it saves us from a different issue (which Yarik will shortly file) which is that search results are reversed.

It will be noted that this PR also includes ability to play video from poster page, which is also a good UX improvement- even if not strictly speaking best practice.

File: src/hooks/posters.ts [hunk 2]
```diff
@@ -75,17 +93,91 @@ export const usePosters = (posterHallId: string) => {
...
+  const liveFilteredPosterVenues = useMemo(
+    () =>
+      liveFilter
+        ? filteredPostersBySubCategory.filter(
+            (filteredPostersBySubCategory) =>
+              filteredPostersBySubCategory.isLive
+          )
+        : filteredPostersBySubCategory,
+    [filteredPostersBySubCategory, liveFilter]
+  );
```
Comment (0xdevalias) [inline src/hooks/posters.ts:180]: There is already filtering going based on 'live' above this in `liveFilterResultsSubCategory` and similar. Why is this also needed here?

Comment (0xdevalias) [reply src/hooks/posters.ts:180]: Additionally, use the guard pattern as described above rather than nesting complex ternaries like this.

File: src/hooks/posters.ts [hunk 3]
```diff
@@ -170,14 +262,22 @@ export const usePosters = (posterHallId: string) => {
...
     liveFilter,
     bookmarkedFilter,
 
     increaseDisplayedPosterCount,
     setSearchInputValue,
     setLiveFilter,
     setBookmarkedFilter,
+    setCategoryFilter,
+    setSubCategoryFilter,
+    unsetCategoryFilter,
```
Comment (0xdevalias) [inline src/hooks/posters.ts:280]: Subcategory is a single word, so shouldn't be cased as 'SubCategory':

- https://en.wikipedia.org/wiki/Subcategory

File: src/settings.ts (hunk 1 above)
Comment (0xdevalias) [inline src/settings.ts:157]: Revert this unrelated change, as it's handled in https://github.com/sparkletown/sparkle/pull/1601

```suggestion
  "image/png,image/x-png,image/gif,image/jpg,image/jpeg,image/tiff,image/bmp,image/gif";
```

File: src/types/posters.ts [hunk 4]
```diff
@@ -1,6 +1,12 @@
+export interface Category {
+  category: string;
+  subcategory: string;
+}
+
 export type Poster = {
   authorName: string;
   categories: string[];
+  subcategories?: Array<Category>;
```
Comment (0xdevalias) [inline src/types/posters.ts:9]: Why does `Category` have both 'category' and 'subcategory' keys on it? And given it's only used in the `Poster`'s `subcategories` key.. should it not be a `Subcategory` at the very least?

Why is `Poster`'s `categories` of type `string[]`, yet `subcategories` is `Array<Category>`?

Why are we using `Array<foo>` instead of our standard `foo[]` syntax?

File: src/types/venues.ts [hunk 5]
```diff
@@ -179,6 +179,7 @@ export interface BaseVenue {
   leftLogo?: string;
   rightLogo?: string;
   canBeBookmarked?: boolean;
+  showCategoriesFilter?: boolean;
```
Comment (0xdevalias) [inline src/types/venues.ts:182]: It would make sense to sort this with all of the other `show*` keys here.

File: src/components/templates/PosterHall/PosterHall.scss [hunk 6]
```diff
@@ -29,4 +31,30 @@
...
+    justify-content: center;
+    margin: $spacing--xs;
+    flex-wrap: wrap;
+  }
+
+  &__subcategory {
+    font-size: $font-size--xxs;
+    cursor: pointer;
+    text-transform: capitalize;
+  }
```
Comment (0xdevalias) [inline src/components/templates/PosterHall/PosterHall.scss:59]: It would probably make sense to extract the categories/subcategories stuff into their own component rather than batching it all together in `PosterHall`?

**Comment by mobilevinay** [comment]
Glenn reviewed this today...
Someone else should look at it... it looks like it could break stuff.

**Comment by mobilevinay** [comment]
CTO approves not accepting these changes.

**Comment by mike-lvov** [comment]
Doesn't seem relevant anymore
